drivers/i2c/i2c_npcx_controller.c

```diff
@@ -433,6 +464,22 @@ static void i2c_ctrl_init_module(const struct device *dev)
 	/* Enable module - before configuring CTL1 */
 	inst->SMBCTL2  |= BIT(NPCX_SMBCTL2_ENABLE);
 
+#ifdef CONFIG_I2C_TARGET
+	volatile uint8_t *reg_smbaddr;
+
+	/* Clear all the SMBnADDR */
+	for (int i = 0; i < NPCX_I2C_FLAG_COUNT; i++) {
+		reg_smbaddr = npcx_i2c_ctrl_target_get_reg_smbaddr(dev, i);
+
+		if (reg_smbaddr == NULL) {
+			LOG_ERR("Invalid SMBnADDR register for target %d", i);
+			continue;
```

Comment:
is `continue` correct here? feels like this should `return` instead

Reply:
unless this is expected for some of the loop iteration, then maybe the message should be a debug level one

Reply:
Yes, we like to iterate to clear SMBADDR1~SMBADDR8 to zero.  
Think further, the check should be not necessary because the value of `i` passed to `npcx_i2c_ctrl_target_get_reg_smbaddr(dev, i);`  is always a valid value within `enum npcx_i2c_flag`
I will update a new patch to remove it.